src/os/bluestore/BlueStore.h

```diff
@@ -2381,6 +2381,11 @@ class BlueStore : public ObjectStore,
     RWLock::WLocker l(debug_read_error_lock);
     debug_mdata_error_objects.insert(o);
   }
+  void compact() override {
+    assert(db);
```

Comment:
Need to see the impact to front end IO's when compaction is progress, from mark's runs we have seen a slowdown in IO's when compaction is in progress. It is better to warn users when issuing the command. We shouldn't be seeing any timeouts because of this invocation.